content/docs/fundamentals/converting/resolving-errors@zh_CN.md

```diff
@@ -0,0 +1,350 @@
+---
+$title: 解决验证错误
+$order: 2
+toc: true
+---
+
+[TOC]
+
+在本部分中，我们将讲解 AMP 网页中的 AMP 验证错误及其解决方法。请注意，这些错误可能会在您的控制台中以不同的顺序出现。
+
+## 包含字符集
+
+首先，我们将修正以下错误：
+
+<pre class="error-text">
+必需的标记“meta charset=utf-8”缺失或不正确。
+</pre>
+
+要正确显示文字，AMP 要求您为网页指定字符集。此外，元字符集信息必须是 `<head>` 标记的第一个子级。之所以必须将该标记作为第一个子级，是为了避免重复解析在元字符集标记前面添加的内容。
+
+将以下代码**添加**为 `<head>` 标记的第一行代码：
+
+
+```html
+<meta charset="utf-8" />
+```
+
+**保存**文件并重新加载网页。验证字符集错误是否不再出现。
+
+## 包含权威链接
+
+现在，我们来看看以下错误：
+
+<pre class="error-text">
+必需的标记“link rel=canonical”缺失或不正确。
+</pre>
+
+每个 AMP 文档都必须拥有引用此文档“权威”版本的链接。在本教程的[使您的网页可被轻松发现](/zh_cn/docs/fundamentals/converting/discoverable.html)步骤中，我们将详细了解权威网页的定义，以及实现权威关联的不同方法。
+
+对于本教程，我们将要转换的原始 HTML 文章视为权威网页。
+
+请将以下代码**添加**到 `<meta charset="utf-8" />` 标记下方：
+
+```html
+<link rel="canonical" href="/article.html">
+```
+
+{% call callout('注意', type='note') %}
+您可以创建独立的权威 AMP 网页。在这种情况下，权威链接仍是必需的，但应指向 AMP 文章本身：
+
+```html
+<link rel="canonical" href="article.amp.html">
+```
+{% endcall %}
+
+现在，**重新加载**网页。虽然仍有很多错误需要修正，但权威链接错误已不复存在。
+
+## 指定 AMP 属性
+
+AMP 要求网页的根 `<html>` 元素中具有相关属性，以将网页声明为 AMP 文档。
+
+<pre class="error-text">
+标记“html ⚡ for top-level html”中缺少必需的属性“⚡”
+必需的标记“html ⚡ for top-level html”缺失或不正确。
+</pre>
+
+只需将 `⚡` 属性添加到 `<html>` 标记，即可解决上述错误，如下所示：
+
+```html
+<html ⚡ lang="en">
+```
+
+现在，请重新加载网页并检查这两个错误是否已消失。
+
+{% call callout('注意', type='note') %}
+虽然建议您指定 `⚡`，但您也可以使用 `amp` 属性来替代 `⚡` 属性，如下所示：
+
+```html
+<html amp lang="en">
+```
+{% endcall %}
+
+## 指定视口
+
+接下来，我们将解决以下错误：
+
+<pre class="error-text">
+必需的标记“meta name=viewport”缺失或不正确。
+</pre>
+
+AMP 要求您为视口定义 `width` 和 `minimum-scale`。这两个值必须分别定义为 `device-width` 和 `1`。视口是 HTML 网页的 `<head>` 中包含的常用标记。
+
+要解决视口错误，请将以下 HTML 代码段添加到 `<head>` 标记中：
+
+```html
+<meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+```
+
+在 AMP 中，必需为 `width` 和 `minimum-scale` 指定值。定义 `initial-scale` 不是一项强制性要求，但移动网站开发者通常都会指定此值，所以我们建议您也这么做。要详细了解视口和自适应设计，请参阅[配置视口](https://developers.google.com/speed/docs/insights/ConfigureViewport)。
+
+和之前一样，**重新加载**网页并检查错误是否已消失。
+
+## 替换外部样式表
+
+以下错误与使用样式表有关：
+
+<pre class="error-text">
```

Comment:
All of these code blocks need to be in EN, so please update with EN translation (https://www.ampproject.org/docs/fundamentals/converting/resolving-errors).